### PR TITLE
scaffold: reorganize components structure

### PR DESCRIPTION
**Pull request by suabochica**
## What this PR do?

Update the component folder structure according Cory House [guidelines](https://twitter.com/housecor/status/1493947989877997571):

### CODE REVIEW

**Comment by john-angel** [comment]
@suabochica a build error with the yarn file was raised here

**Review by Jkierem — COMMENTED** [review]
Rather than saying that 'we use Cory's guidelines', I would say 'we are based on Cory's guidelines' and remove the module.css entirely. Also I would suggest exporting the types in the index too.

**Review by john-angel — APPROVED** [review]
Looks good!

File: README.md [hunk 1]
```diff
@@ -1,16 +1,51 @@
...
+- Jest v27.4.3
+
+## 📁 Folder Structure
+ we are based on Cory's [guidelines](https://twitter.com/housecor/status/1493947989877997571?s=20&t=47Knp8NaM3KOzADGxkTH0w).
+
+1. Button.tsx - component
+2. Button.types.ts - TS types
+3. Button.test.ts - Jest + testing-library unit tests
+4. Button.stories.tsx - Storybook stories
+6. index.tsx - Re-exports ([barrel](https://trusz.github.io/posts/barrel-pattern-in-typescript/))
```
Comment (Jkierem) [inline README.md]: Numbering. Should be 5.

File: src/components/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export { default as Datepicker } from './Datepicker/Datepicker';
+export { default as FieldContainer } from './FieldContainer/FieldContainer';
+export { default as FlexLayout } from './FlexLayout/FlexLayout';
+export { default as Form } from './Form/Form';
+export { default as NativeDatepicker } from './NativeDatepicker/NativeDatepicker';
+export { default as Select } from './Select/Select';
+export { default as Tag } from './Tag/Tag';
+export { default as TextArea } from './TextArea/TextArea';
+export { default as TextField } from './TextField/TextField';
+export { default as Toggle } from "./Toggle/Toggle";
```
Comment (Jkierem) [inline src/components/index.ts]: This file needs to be modified to match the new structure. Should export everything on the barrels (index.ts files) and let's also make sure we don't use default exports in the barrels.

File: README.md [hunk 3]
```diff
@@ -24,16 +59,32 @@ import Button from '@e-toast/lib/Button';
```
Comment (Jkierem) [inline README.md:60]: The usage section needs to be updated. I believe single component imports now become `import { Button } from '@e-toast/lib/components/Button'`. Please check it by running a build and checking the lib folder matches the expected structure.

**Review by Jkierem — COMMENTED** [review]
There are some details that need addressing. There is a chance that a follow up to this change is needed. One more requirement to the component structure I would add is stop using default exports on component.tsx files and change them for named exports.

File: src/components/index.ts (hunk 2 above)
Comment (suabochica) [reply src/components/index.ts]: So this file should be something like this:

```
export * from './Tag/Tag';
export * from './TextArea/TextArea';
export * from './TextField/TextField';
export * from "./Toggle/Toggle";
```

Comment (Jkierem) [reply src/components/index.ts]: More like

```
export * from './Tag';
export * from './TextArea';
export * from './TextField';
export * from "./Toggle";
```

**Review by Jkierem — COMMENTED** [review]
Please update the package.json. the "main" should be changed to point to "lib/components/index.js". This is a critical change. This is so imports from "@e-toast/react" work as intended

File: tsconfig.json [hunk 4]
```diff
@@ -31,8 +31,8 @@
...
     "**/*.stories.tsx",
-    "./src/setupTests.ts",
-    "./stories/**"
+    "src/**",
+    "src/commons/stories/**"
   ],
-  "files": ["./src/index.ts"]
-}
\ No newline at end of file
+  "files": ["src/components/index.ts"]
```
Comment (Jkierem) [inline tsconfig.json]: This configuration causes the build to be missing all the component barrels. A better config would be to remove `files` and have  the `include` option be `["src/components/**/index.tsx", "src/components/index.ts"]`

File: src/components/FlexLayout/FlexLayout.options.tsx [hunk 5]
```diff
@@ -1,5 +1,6 @@
-export const AlignItems = [
```
Comment (Jkierem) [inline src/components/FlexLayout/FlexLayout.options.tsx:1]: Is this component missing a index.tsx file?

File: src/components/Form/Form.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
 import React, { useEffect, useMemo, useReducer } from 'react';
```
Comment (Jkierem) [inline src/components/Form/Form.tsx:1]: (I can't leave a comment on an empty file so leaving it here) The index.tsx file is empty

**Review by Jkierem — COMMENTED** [review]
Great work! This is a huge PR and it took me a while to go over everything. Sorry for all the comments but there are a lot of things to check so the library doesn't break.

File: README.md (hunk 3 above)
Comment (suabochica) [reply README.md:60]: Done!

File: tsconfig.json (hunk 4 above)
Comment (suabochica) [reply tsconfig.json]: Nice catch! Done 🙇🏾

File: src/components/FlexLayout/FlexLayout.options.tsx (hunk 5 above)
Comment (suabochica) [reply src/components/FlexLayout/FlexLayout.options.tsx:1]: 🙈

File: src/components/Form/Form.tsx (hunk 6 above)
Comment (suabochica) [reply src/components/Form/Form.tsx:1]: My bad 😛

**Comment by suabochica** [comment]
Not big deal, Actually thanks for all your comments, they boarded several scenarios that I was omitting in the beginning 🙇🏾

**Pull request merged**